lib/ex_unit/lib/ex_unit/server.ex

```diff
@@ -116,11 +142,29 @@ defmodule ExUnit.Server do
 
   def handle_info({:DOWN, ref, _, _, _}, config) do
     config = remove_log_capture(ref, config)
+    config = release_device(ref, config)
     {:noreply, config}
   end
 
-  def handle_info(_msg, state) do
-    {:noreply, state}
+  def handle_info(msg, state) do
+    super(msg, state)
+  end
+
+  defp release_device(ref, %{captured_devices: {devices, refs}} = config) do
+    case Map.pop(refs, ref) do
+      {{device, pid}, refs} ->
+        devices = HashSet.delete(devices, device)
+        Process.demonitor(ref, [:flush])
+        try do
+          Process.unregister(device)
+        rescue
+          ArgumentError -> :ok
+        after
+          Process.register(pid, device)
```

Comment:
If register raises here another process has registered as `device`. It is not the `string_io` process so we can rescue `ArgumentError` and ignore it.